engine/apps/alerts/models/channel_filter.py

```diff
@@ -172,14 +172,8 @@ def slack_channel_slack_id(self) -> typing.Optional[str]:
         return self.slack_channel.slack_id if self.slack_channel else None
 
     @property
-    def slack_channel_id_or_org_default_id(self):
-        organization = self.alert_receive_channel.organization
-
-        if organization.slack_team_identity is None:
-            return None
```

Comment:
I'm not sure this check needs to be here? 🤔

```python
if organization.slack_team_identity is None:
  return None
``` 

`organization.slack_team_identity` and `organization.default_slack_channel` are two distinct things.

Also, I'm _fairly_ certain that when a user uninstalls the Slack integration (ie. `organization.slack_team_identity` is set to `None`) we unset the `default_slack_channel` as well:

https://github.com/grafana/oncall/blob/jorlando/cleanup_slack_id/engine/apps/slack/installation.py#L70

https://github.com/grafana/oncall/blob/jorlando/cleanup_slack_id/engine/apps/slack/tasks.py#L200

---

I noticed we use `if organization.slack_team_identity is not None`, throughout the codebase, to basically mean "does the org have Slack connected?".

I was thinking of adding this to the `Organization` model:

```python
@property
def is_slack_connected(self) -> None:
  return self.slack_team_identity is not None
```

but if we feel that is appropriate, it should _probably_ be refactored in a separate PR

Reply:
(I also refactored the two references to `slack_channel_id_or_org_default_id` to use `slack_channel_or_org_default`)